commons/zenoh-protocol/src/core/encoding.rs

```diff
@@ -135,20 +139,18 @@ impl Encoding {
         if suffix.as_ref().is_empty() {
             Some(Encoding::Exact(prefix))
         } else {
-            Some(Encoding::WithSuffix(prefix, suffix.into()))
+            Some(Encoding::WithSuffix(prefix, suffix.into().into()))
```

Comment:
The double `into()` without annotation is a bit weird to me. What are the conversions steps?

Reply:
Cow<str> is used as intermediate to maintain the previous API contract and legibility

Reply:
Then I just wonder how the compiler manages to correctly infer that... I wasn't expecting this to work :)

Reply:
It's thanks to the function's bounds: the only thing it knows about `suffix` here is that it can be `into()`ed into a `Cow<str>`, so that's the only thing the first `into()` can do. The second into knows the slot expects a `CowStr`, and boom, magic :D

Reply:
That's right, I have overlooked the function bounds for a moment. It LGTM.